erts/configure.in

```diff
@@ -2858,22 +2883,115 @@ if test X${enable_hipe} != Xno; then
            #error "HiPE does not work without a libc that can guarantee that sigaltstack works"
            #endif	/* !(__GLIBC__ || __DARWIN__ || __NetBSD__ || __FreeBSD__ || __sun__) */],
         [AC_MSG_RESULT([yes])],
-        [enable_hipe=no
-         AC_MSG_RESULT([no, musl probably used. Need glibc to work properly])
-         AC_MSG_WARN([HiPE disabled due to lack of safe signal delivery])])
+        [if test X${enable_hipe} = Xyes; then
+           AC_MSG_ERROR([HiPE needs glibc to work properly])
+         else
+           enable_hipe=no
+           AC_MSG_RESULT([no, musl probably used. Need glibc to work properly])
+           AC_MSG_WARN([HiPE disabled due to lack of safe signal delivery])
+         fi])
+       case "$ARCH-$OPSYS" in
+	    x86-linux|amd64-linux|x86-darwin*|amd64-darwin*|ppc-linux|ppc64-linux|ppc-darwin|arm-linux|amd64-freebsd|x86-freebsd|x86-sol2|amd64-sol2|ultrasparc-linux)
+              ;;
+            *)
+              if test X${enable_hipe} = Xyes; then
+                AC_MSG_ERROR([HiPE is not supported on $ARCH-$OPSYS])
+              else
+                enable_hipe=no
+                AC_MSG_WARN([HiPE disabled due to lack to support on $ARCH-$OPSYS])
+              fi
+              ;;
+        esac
+fi
+
+dnl Test if BEAMASM can be enabled
+if test ${enable_beamasm} != no; then
+   case "$ARCH" in
+        amd64)
+           ;;
+        *)
+           if test ${enable_beamasm} = yes; then
+             AC_MSG_ERROR([BEAMASM only works on x86 64-bit])
+           else
+             enable_beamasm=no
+             AC_MSG_WARN([BEAMASM disabled due to lack to support on $ARCH-$OPSYS])
+           fi
+           ;;
+   esac
+
+   if test ${enable_beamasm} != no; then
+       if test "$CXX" != false; then
+          AC_LANG_PUSH(C++)
+          old_CXXFLAGS=$CXXFLAGS
+          CXXFLAGS="$CXXFLAGS -std=c++17"
```

Comment:
I suspect this directive causes a compilation error on macOS Xcode clang compiler.
See https://bugs.erlang.org/browse/ERL-1346 for the details.

Reply:
You can try -std=c++1z instead.

Reply:
Setting C and C++ compiler with different names solved the problem. My bad. The correct names of CC and CXX are:
`CC=/usr/local/opt/ccache/libexec/clang CXX=/usr/local/opt/ccache/libexec/clang++`
ERL-1346 set to "resolved" as well.
BTW thanks for the `c++1z` info, @tschuett!